src/sdkClient/clientInputValidation.ts

```diff
@@ -30,21 +30,27 @@ export function clientInputValidationDecorator<TClient extends SplitIO.IClient |
   /**
    * Avoid repeating this validations code
    */
-  function validateEvaluationParams(maybeKey: SplitIO.SplitKey, maybeFeatureFlagNameOrNames: string | string[], maybeAttributes: SplitIO.Attributes | undefined, methodName: string) {
+  function validateEvaluationParams(maybeKey: SplitIO.SplitKey, maybeFeatureFlagNameOrNames: string | string[] | undefined, maybeAttributes: SplitIO.Attributes | undefined, methodName: string, maybeFlagSetNameOrNames?: string | string[] | undefined) {
     const multi = startsWith(methodName, 'getTreatments');
     const key = validateKey(log, maybeKey, methodName);
-    const splitOrSplits = multi ? validateSplits(log, maybeFeatureFlagNameOrNames, methodName) : validateSplit(log, maybeFeatureFlagNameOrNames, methodName);
+    let splitOrSplits: string | string[] | false = false;
+    if (maybeFeatureFlagNameOrNames) {
+      splitOrSplits = multi ? validateSplits(log, maybeFeatureFlagNameOrNames, methodName) : validateSplit(log, maybeFeatureFlagNameOrNames, methodName);
+    }
     const attributes = validateAttributes(log, maybeAttributes, methodName);
     const isNotDestroyed = validateIfNotDestroyed(log, readinessManager, methodName);
+    // @todo maybeFlagSets validation method
+    const flagSetOrFlagSets = maybeFlagSetNameOrNames;
```

Comment:
You can reuse `validateSplits` to validate flag sets, in order to avoid increasing the code size. Check `validateSplits` and you will see that you can pass additional params (listName and item) to customize the logging  (should log "... flag set" rather than "... feature flag name"). 

You can also simplify the `maybeFlagSetNameOrNames` param as `string[] | undefined` (e.g., for `getTreatmentsByFlagSets` you can call it with `[maybeFlagSet]`). This way you don't need to use the `multi` variable here, and only call `validateSplits` (the plural version of `validateSplit`), of course with listName and item params updated to log errors and warnings properly.

Reply:
I will add input validation in another pr, but I will take this into account